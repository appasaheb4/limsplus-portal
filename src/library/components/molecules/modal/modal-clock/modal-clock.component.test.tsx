import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import {ModalClock} from './Modal-clock.component';

describe('ModalClock component', () => {
  it('render modalClock correctly', () => {
    const modalClock = render(
      <ModalClock
        onClick={() => jest.fn()}
        onClose={() => jest.fn()}
        show={true}
      />,
    );
    expect(modalClock).toMatchSnapshot();
  });
});
