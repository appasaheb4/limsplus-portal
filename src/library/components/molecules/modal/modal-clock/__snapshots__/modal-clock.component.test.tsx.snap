// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render modalClock correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="container"
        >
          <div
            class="justify-center items-center  overflow-x-hidden overflow-y-auto fixed inset-0 z-50 outline-none focus:outline-none"
          >
            <div
              class="relative w-0 my-6 mx-auto"
            >
              <div
                class="border-0 justify-center items-center rounded-lg shadow-lg relative flex flex-col  bg-white outline-none focus:outline-none"
              >
                <div
                  class="relative p-2 flex-auto"
                >
                  <div
                    class="react-timekeeper css-3g92bx-timepicker"
                  >
                    <div
                      class="react-timekeeper__top-bar css-hcsisw-wrapper"
                      data-testid="topbar"
                    >
                      <div
                        class="react-timekeeper__tb-minute-wrapper css-13gu23t-hourWrapper"
                      >
                        <span
                          class="react-timekeeper__tb-hour react-timekeeper__tb-hour--active css-1igvx5z-timeSelected-time-TopBar"
                          data-testid="topbar_hour"
                        >
                          12
                        </span>
                      </div>
                      <span
                        class="react-timekeeper__tb-colon css-1lnifv0-colon"
                      >
                        :
                      </span>
                      <div
                        class="react-timekeeper__tb-hour-wrapper css-6bnttf-minuteWrapper"
                      >
                        <span
                          class="react-timekeeper__tb-minute  css-nntsr1-time"
                          data-testid="topbar_minute"
                        >
                          30
                        </span>
                      </div>
                      <button
                        class="react-timekeeper-button-reset react-timekeeper__tb-meridiem css-1outj7w-meridiem"
                        data-testid="topbar_meridiem"
                        name="meridiem"
                        type="button"
                      >
                        pm
                      </button>
                    </div>
                    <div
                      class="react-timekeeper__clock-wrapper css-1157z6n-style"
                      data-testid="clock-wrapper"
                    >
                      <div
                        class="react-timekeeper__clock css-cowp6p-styles"
                      >
                        <div
                          class="react-timekeeper__clock-hours css-m7oa3k-numbersWrapperStyle"
                        >
                          <span
                            class="css-1ctvv7l-numbersStyle"
                            data-testid="number_hour_outer"
                            style="transform: translate(147.99999999999997px, -2.2627944162882727px);"
                          >
                            1
                          </span>
                          <span
                            class="css-1ctvv7l-numbersStyle"
                            data-testid="number_hour_outer"
                            style="transform: translate(188.26279441628822px, 37.99999999999995px);"
                          >
                            2
                          </span>
                          <span
                            class="css-1ctvv7l-numbersStyle"
                            data-testid="number_hour_outer"
                            style="transform: translate(203px, 92.99999999999999px);"
                          >
                            3
                          </span>
                          <span
                            class="css-1ctvv7l-numbersStyle"
                            data-testid="number_hour_outer"
                            style="transform: translate(188.26279441628824px, 148px);"
                          >
                            4
                          </span>
                          <span
                            class="css-1ctvv7l-numbersStyle"
                            data-testid="number_hour_outer"
                            style="transform: translate(148.00000000000006px, 188.26279441628822px);"
                          >
                            5
                          </span>
                          <span
                            class="css-1ctvv7l-numbersStyle"
                            data-testid="number_hour_outer"
                            style="transform: translate(93.00000000000003px, 203px);"
                          >
                            6
                          </span>
                          <span
                            class="css-1ctvv7l-numbersStyle"
                            data-testid="number_hour_outer"
                            style="transform: translate(38px, 188.26279441628824px);"
                          >
                            7
                          </span>
                          <span
                            class="css-1ctvv7l-numbersStyle"
                            data-testid="number_hour_outer"
                            style="transform: translate(-2.262794416288216px, 148.00000000000006px);"
                          >
                            8
                          </span>
                          <span
                            class="css-1ctvv7l-numbersStyle"
                            data-testid="number_hour_outer"
                            style="transform: translate(-17px, 93.00000000000003px);"
                          >
                            9
                          </span>
                          <span
                            class="css-1ctvv7l-numbersStyle"
                            data-testid="number_hour_outer"
                            style="transform: translate(-2.262794416288301px, 38.000000000000085px);"
                          >
                            10
                          </span>
                          <span
                            class="css-1ctvv7l-numbersStyle"
                            data-testid="number_hour_outer"
                            style="transform: translate(38.00000000000002px, -2.2627944162882727px);"
                          >
                            11
                          </span>
                          <span
                            class="css-1ctvv7l-numbersStyle"
                            data-testid="number_hour_outer"
                            style="transform: translate(92.99999999999996px, -17px);"
                          >
                            12
                          </span>
                        </div>
                        <svg
                          class="react-timekeeper__clock-hand"
                          height="220"
                          viewBox="0 0 220 220"
                          width="220"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            transform="rotate(360 110 110)"
                          >
                            <line
                              class="react-timekeeper__clock-hand css-yxfhll-line"
                              data-testid="clock-hand"
                              stroke-width="1"
                              x1="110"
                              x2="110"
                              y1="110"
                              y2="22"
                            />
                            <circle
                              class="react-timekeeper__hand-circle-center css-uby47j-centerCircle"
                              cx="110"
                              cy="110"
                              r="1.5"
                            />
                            <circle
                              class="react-timekeeper__hand-circle-outer css-200z2t-outerCircle"
                              cx="110"
                              cy="22"
                              r="17"
                            />
                          </g>
                        </svg>
                      </div>
                      <div
                        class="css-11koacj-meridiemWrapper"
                      >
                        <button
                          class="react-timekeeper-button-reset react-timekeeper__meridiem-toggle  css-1lf0io6-meridiem"
                          data-testid="meridiem_am"
                          type="button"
                        >
                          AM
                        </button>
                        <button
                          class="react-timekeeper-button-reset react-timekeeper__meridiem-toggle react-timekeeper__meridiem--active css-dlroum-meridiem"
                          data-testid="meridiem_pm"
                          type="button"
                        >
                          PM
                        </button>
                      </div>
                    </div>
                    <span
                      class="react-timekeeper__done-button css-yo7vjg-doneButton"
                      data-testid="done-button"
                    >
                      Done
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="opacity-25 fixed inset-0 z-40 bg-black"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="container"
      >
        <div
          class="justify-center items-center  overflow-x-hidden overflow-y-auto fixed inset-0 z-50 outline-none focus:outline-none"
        >
          <div
            class="relative w-0 my-6 mx-auto"
          >
            <div
              class="border-0 justify-center items-center rounded-lg shadow-lg relative flex flex-col  bg-white outline-none focus:outline-none"
            >
              <div
                class="relative p-2 flex-auto"
              >
                <div
                  class="react-timekeeper css-3g92bx-timepicker"
                >
                  <div
                    class="react-timekeeper__top-bar css-hcsisw-wrapper"
                    data-testid="topbar"
                  >
                    <div
                      class="react-timekeeper__tb-minute-wrapper css-13gu23t-hourWrapper"
                    >
                      <span
                        class="react-timekeeper__tb-hour react-timekeeper__tb-hour--active css-1igvx5z-timeSelected-time-TopBar"
                        data-testid="topbar_hour"
                      >
                        12
                      </span>
                    </div>
                    <span
                      class="react-timekeeper__tb-colon css-1lnifv0-colon"
                    >
                      :
                    </span>
                    <div
                      class="react-timekeeper__tb-hour-wrapper css-6bnttf-minuteWrapper"
                    >
                      <span
                        class="react-timekeeper__tb-minute  css-nntsr1-time"
                        data-testid="topbar_minute"
                      >
                        30
                      </span>
                    </div>
                    <button
                      class="react-timekeeper-button-reset react-timekeeper__tb-meridiem css-1outj7w-meridiem"
                      data-testid="topbar_meridiem"
                      name="meridiem"
                      type="button"
                    >
                      pm
                    </button>
                  </div>
                  <div
                    class="react-timekeeper__clock-wrapper css-1157z6n-style"
                    data-testid="clock-wrapper"
                  >
                    <div
                      class="react-timekeeper__clock css-cowp6p-styles"
                    >
                      <div
                        class="react-timekeeper__clock-hours css-m7oa3k-numbersWrapperStyle"
                      >
                        <span
                          class="css-1ctvv7l-numbersStyle"
                          data-testid="number_hour_outer"
                          style="transform: translate(147.99999999999997px, -2.2627944162882727px);"
                        >
                          1
                        </span>
                        <span
                          class="css-1ctvv7l-numbersStyle"
                          data-testid="number_hour_outer"
                          style="transform: translate(188.26279441628822px, 37.99999999999995px);"
                        >
                          2
                        </span>
                        <span
                          class="css-1ctvv7l-numbersStyle"
                          data-testid="number_hour_outer"
                          style="transform: translate(203px, 92.99999999999999px);"
                        >
                          3
                        </span>
                        <span
                          class="css-1ctvv7l-numbersStyle"
                          data-testid="number_hour_outer"
                          style="transform: translate(188.26279441628824px, 148px);"
                        >
                          4
                        </span>
                        <span
                          class="css-1ctvv7l-numbersStyle"
                          data-testid="number_hour_outer"
                          style="transform: translate(148.00000000000006px, 188.26279441628822px);"
                        >
                          5
                        </span>
                        <span
                          class="css-1ctvv7l-numbersStyle"
                          data-testid="number_hour_outer"
                          style="transform: translate(93.00000000000003px, 203px);"
                        >
                          6
                        </span>
                        <span
                          class="css-1ctvv7l-numbersStyle"
                          data-testid="number_hour_outer"
                          style="transform: translate(38px, 188.26279441628824px);"
                        >
                          7
                        </span>
                        <span
                          class="css-1ctvv7l-numbersStyle"
                          data-testid="number_hour_outer"
                          style="transform: translate(-2.262794416288216px, 148.00000000000006px);"
                        >
                          8
                        </span>
                        <span
                          class="css-1ctvv7l-numbersStyle"
                          data-testid="number_hour_outer"
                          style="transform: translate(-17px, 93.00000000000003px);"
                        >
                          9
                        </span>
                        <span
                          class="css-1ctvv7l-numbersStyle"
                          data-testid="number_hour_outer"
                          style="transform: translate(-2.262794416288301px, 38.000000000000085px);"
                        >
                          10
                        </span>
                        <span
                          class="css-1ctvv7l-numbersStyle"
                          data-testid="number_hour_outer"
                          style="transform: translate(38.00000000000002px, -2.2627944162882727px);"
                        >
                          11
                        </span>
                        <span
                          class="css-1ctvv7l-numbersStyle"
                          data-testid="number_hour_outer"
                          style="transform: translate(92.99999999999996px, -17px);"
                        >
                          12
                        </span>
                      </div>
                      <svg
                        class="react-timekeeper__clock-hand"
                        height="220"
                        viewBox="0 0 220 220"
                        width="220"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          transform="rotate(360 110 110)"
                        >
                          <line
                            class="react-timekeeper__clock-hand css-yxfhll-line"
                            data-testid="clock-hand"
                            stroke-width="1"
                            x1="110"
                            x2="110"
                            y1="110"
                            y2="22"
                          />
                          <circle
                            class="react-timekeeper__hand-circle-center css-uby47j-centerCircle"
                            cx="110"
                            cy="110"
                            r="1.5"
                          />
                          <circle
                            class="react-timekeeper__hand-circle-outer css-200z2t-outerCircle"
                            cx="110"
                            cy="22"
                            r="17"
                          />
                        </g>
                      </svg>
                    </div>
                    <div
                      class="css-11koacj-meridiemWrapper"
                    >
                      <button
                        class="react-timekeeper-button-reset react-timekeeper__meridiem-toggle  css-1lf0io6-meridiem"
                        data-testid="meridiem_am"
                        type="button"
                      >
                        AM
                      </button>
                      <button
                        class="react-timekeeper-button-reset react-timekeeper__meridiem-toggle react-timekeeper__meridiem--active css-dlroum-meridiem"
                        data-testid="meridiem_pm"
                        type="button"
                      >
                        PM
                      </button>
                    </div>
                  </div>
                  <span
                    class="react-timekeeper__done-button css-yo7vjg-doneButton"
                    data-testid="done-button"
                  >
                    Done
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="opacity-25 fixed inset-0 z-40 bg-black"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
